packages/wrangler/src/__tests__/dev.test.tsx

```diff
@@ -231,6 +231,82 @@ describe("Dev component", () => {
       );
     });
   });
+
+  describe("custom builds", () => {
+    it("should run a custom build before starting `dev`", async () => {
+      writeWranglerToml({
+        build: {
+          command: `node -e "console.log('custom build'); require('fs').writeFileSync('index.js', 'export default { fetch(){ return new Response(123) } }')"`,
+        },
+      });
+
+      await runWrangler("dev index.js");
```

Comment:
No, because we use the mocked version of the component that immediately exits (https://github.com/cloudflare/wrangler2/blob/b6f226624417ffa4b5e7c3098d4955bc23d58603/packages/wrangler/src/__tests__/jest.setup.ts#L59-L69)